crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -245,46 +235,51 @@ impl<T: StateRead + ?Sized> StateReadExt for T {}
 #[async_trait]
 pub(crate) trait StateWriteExt: StateWrite {
     #[instrument(skip_all, fields(address = %address.display_address(), %asset, balance), err)]
-    fn put_account_balance<TAddress, TAsset>(
+    fn put_account_balance<'a, TAddress, TAsset>(
         &mut self,
-        address: TAddress,
-        asset: TAsset,
+        address: &TAddress,
+        asset: &'a TAsset,
```

Comment:
Gonna ask this here because I have noticed this throughout the PR: what required changing this from `TAsset` and `TAddress` to take their borrows, respectively? If I am not mistaken these methods allowed being called with `&asset` and `&address` already.

Reply:
They did, but they also allowed for passing by value.  I wanted to disallow the ability to pass by value in these cases, since it would be easy to accidentally copy the data.

Reply:
OK, cool. I figured that would be it.